app/services/cluster_stats/update_epoch_duration_service.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module ClusterStats
+  class UpdateEpochDurationService
+
+    AVG_EPOCH_SIZE = 5
+
+    def initialize(network)
+      @network = network
+    end
+
+    def call
+      cluster_stat.update_attribute(:epoch_duration, epoch_duration)
+
+      cluster_stat
+    end
+
+    private
+
+    attr_reader :network
+
+    def cluster_stat
+      ClusterStat.find_or_create_by(network: network)
+    end
+
+    def epochs
+      @epochs ||= EpochWallClock.by_network(network).last(AVG_EPOCH_SIZE)
```

Comment:
by_network scope sorts epochs in descending order, so you should take first 5 records, not last. Please check how it used in other places of the app next time.